### ISSUE TITLE
Prevent webmentions sending for specified links

### ISSUE THREAD

**Issue by colin-walker** [pfefferle/wordpress-webmention#166]
The webmention plugin sends mention for all links in a post but there may be times when you don’t want a webmention sent, especially when dealing with social properties like micro.blog or Mastodon where a mention may create duplication if your post is going to be POSSE’d over.

How about a mechanism to specify on a link that you don’t want it to trigger a mention?

A site blacklist is too extreme as this precludes ever sending to that domain so here’s a suggestion:

We could add `class=“nomention”` to the link then check for this in the plugin using the following:

Replace `$links = wp_extract_urls( $post->post_content );` within `public static function send_webmentions` with

```
preg_match_all('/<a[^>]+>/i',$post->post_content, $results); 

$mentions = '';
foreach ($results[0] as $link) {
  if (!strpos($link, 'nomention')) {
    $mentions .= $link;
  }
}

$links = wp_extract_urls($mentions);
```

to cycle through the <a> tags in the post, add those _without_ the nomention class to a new string and extract the URLs from that strong instead.

**Comment by dshanske** [comment]
Isn't this basically rel=nofollow ?

**Comment by pfefferle** [comment]
This limits the parser to links and ignores images, videos, ... It is also not very user friendly, because nearly noone knows about microformats and writes plain HTML. Besides that, I do not think this is really an issue that needs to be solved on the WordPress side, only if micro.blog does not handle the Webmentions well.

**Comment by colin-walker** [comment]
Yes, I was thinking about it like nofollow but that doesn’t impact webmentions does it?

Good point, about images & videos - that could be extended.

It was just a quick thought about generating a mechanism to do it.

**Comment by pfefferle** [comment]
For my better understanding, is this really an issue or is it only because of the handling of micro.blog?

**Comment by colin-walker** [comment]
The micro.blog incident was what triggered the thinking but surely they may be other times when you don’t necessarily want/need a mention generated. Just thinking out loud.

**Comment by pfefferle** [comment]
I would prefer a more [gen 2 - 4](https://indieweb.org/generations) friendly solution, if this is really a problem that needs to be solved.

**Comment by pfefferle** [comment]
@colin-walker but is this really link specific, or is a handling on blog level also a solution?

**Comment by colin-walker** [comment]
As I say, it was just an initial idea based on the plugin itself. If there was a wider need then, yes, there would no doubt be a better way of implementing it at the spec level.

**Comment by pfefferle** [comment]
sorry, I meant "post-level" instead of "blog-level"...

Perhaps we can add a "flag" to the editor, to disable "send pings" for specific posts.

**Comment by dshanske** [comment]
There already is a low level solution. Filter them..

https://github.com/pfefferle/wordpress-webmention/blob/master/includes/class-webmention-sender.php#L154

**Comment by dshanske** [comment]
But you can always add something to the box that sends pings and trackbacks on a post a blacklist hooked into this filter

**Comment by colin-walker** [comment]
Thanks David. I think I've got it using the filter hook.

https://github.com/colin-walker/nomention

I know this is an edge case but it was something I wanted for my own workflow and the filter lets me do it without having to edit the plugin. Appreciate the pointer.

**Comment by dshanske** [comment]
Any time

**Comment by snarfed** [comment]
I'd love this too! I don't quite need it per post per link, either one would be fine for me personally, either checkbox per post or maybe global set of URLs/domains to never send wms too.
